developers/weaviate/client-libraries/typescript/v2_v3_migration.md

```diff
@@ -129,132 +101,308 @@ Once it has been instantiated, you will notice that the client API is different
 ## Major changes
 
 From a user's perspective, major changes with the `v3` client include:
- 
 
 ### Better Typescript Support
 
-  - generics, functions and methods
-
-The `v3` client introduces an extensive set of helper classes to interact with Weaviate. These classes are used to provide strong typing, and to make the client more user-friendly such as through IDE autocompletion.
+The `v3` client introduces better Typescript support. A lot of these benefits are visible with better IDE intellisense and type-safety. To add to this, we've made it possible for users to define their own Generics.
 
-Take a look at the examples below.
+<Tabs groupId="languages">
+<TabItem value="generics" label="Define Generics">
 
-import QuickStartCode from '!!raw-loader!/_includes/code/graphql.filters.nearText.generic.py';
+```ts
+import weaviate from 'weaviate-client';
 
-<Tabs groupId="languages">
-<TabItem value="create" label="Create a collection">
+type Article = {
+  title: string;
+  body: string;
+  wordcount: number;
+}
 
-  <FilteredTextBlock
-    text={PythonCode}
-    startMarker="# START CreateCollectionExample"
-    endMarker="# END CreateCollectionExample"
-    language="py"
-  />
+const collection = client.collections.get<Article>('Article');
+```
 
 </TabItem>
-<TabItem value="query" label="NearText query">
+<TabItem value="query" label="Insert Data">
 
-  <FilteredTextBlock
-    text={QuickStartCode}
-    startMarker="# NearTextExample"
-    endMarker="# END NearTextExample"
-    language="py"
-  />
+```ts
+const collection = client.collections.get<Article>('Article');
+
+await collection.insert({ // compiler error since 'body' field is missing in '.insert'
+  title: 'TS is awesome!',
+  wordcount: 9001
+})
+```
 
 </TabItem>
 </Tabs>
 
-In both of these examples, you can see how the helper classes and methods abstract away the need for manual JSONs or strings.
+As in the example above, we can extend the use of Generics to create type-safe operations around collection creation, querying and deletion to mention a few. 
```

Comment:
1st person, capitalization - Consider:

The example uses generics to create type-safe operations. Use generics to write type-safe methods for other operations such as collection creation, querying, and deletion.